src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -51,21 +51,26 @@ public FObject put_(X x, FObject obj) {
     User        user    = (User) x.get("user");
     AuthService auth    = (AuthService) x.get("auth");
 
-    User toPut = (User) obj;
-    if ( toPut.getId() < 1000 ) {
-      throw new RuntimeException("Unable to update user");
-    }
+     User toPut = (User) obj;
+
+     if ( toPut == null ) {
+       throw new RuntimeException("User can't be null");
+     }
 
-    if ( toPut != null && ! SafetyUtil.equals(toPut.getId(), user.getId()) &&
+     if ( isExistingUser(toPut) &&  toPut.getId() < 1000 ) {
+       throw new RuntimeException("Unable to update user");
+     }
+
+    if ( ! SafetyUtil.equals(toPut.getId(), user.getId()) &&
       ! auth.check(x, GLOBAL_USER_UPDATE) &&
       ! auth.check(x, GLOBAL_SPID_UPDATE) &&
       ! auth.check(x, "spid.update." + user.getSpid()) ) {
       throw new RuntimeException("Unable to update user");
     }
 
     // set spid if not set
-    if ( SafetyUtil.isEmpty((String) toPut.getSpid()) &&
-        ! SafetyUtil.isEmpty((String) user.getSpid()) ) {
+    if ( SafetyUtil.isEmpty(toPut.getSpid()) &&
+        ! SafetyUtil.isEmpty(user.getSpid()) ) {
```

Comment:
Can we also log a warning when the users spid is empty, I don't see this as a valid scenario.